### PR TITLE
Updated integrationtests.yml with new changes

### PR DESCRIPTION
**Pull request by santhiya-mj**

### CODE REVIEW

File: .github/workflows/integrationtests.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ jobs:
   test:
     runs-on: ubuntu-latest
 
+    env:
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:13]: This change is not required

    env:
      CHAT_SERVICE_PORT: 8000
      OLLAMA_PORT: 11434


Remove it

File: .github/workflows/integrationtests.yml [hunk 2]
```diff
@@ -10,6 +10,10 @@ jobs:
...
+      CHAT_SERVICE_PORT: 8000
+      OLLAMA_PORT: 11434
+
     steps:
       # Step 1: Check out the pull request code
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:18]: Use a step to load the .env file

File: .github/workflows/integrationtests.yml [hunk 3]
```diff
@@ -58,12 +62,12 @@ jobs:
             fi
           done
           echo "Error: Server did not start within 2 minutes."
-          exit 1
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:61]: Lets not have this change which is unnecessary

File: .github/workflows/integrationtests.yml [hunk 4]
```diff
@@ -104,6 +108,6 @@ jobs:
 
       # Step 9: Shut down Docker Compose (clean up)
       - name: Docker-compose down
-        working-directory: src/backend  # Specify working directory here as well
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:107]: Lets not have this unnecessary change

File: .github/workflows/integrationtests.yml [hunk 5]
```diff
@@ -104,6 +108,6 @@ jobs:
...
+        working-directory: src/backend
         if: always()
-        run: docker-compose down
```
Comment (SaravanakumarR2018) [inline .github/workflows/integrationtests.yml:109]: Lets not have this unnecessary change

File: build/kill_server.sh [hunk 6]
```diff
@@ -13,7 +13,57 @@ echo "Project root: $PROJECT_ROOT"
 export PROJECT_ROOT
 echo "PROJECT_ROOT is set to: $PROJECT_ROOT"
 
+# Set fallback values for critical environment variables
```
Comment (SaravanakumarR2018) [inline build/kill_server.sh:16]: Add a command to load it from .env file instead of the below lines

# Set fallback values for critical environment variables
export CHAT_SERVICE_PORT=${CHAT_SERVICE_PORT:-8000}
export DB_USER=${DB_USER:-root}
export DB_PASSWORD=${DB_PASSWORD:-admin123}
export DB_HOST=${DB_HOST:-mysql_db}
export MYSQL_DATABASE=${MYSQL_DATABASE:-mydatabase}
export MYSQL_PORT=${MYSQL_PORT:-3306}
export MINIO_ROOT_USER=${MINIO_ROOT_USER:-minioadmin}
export MINIO_ROOT_PASSWORD=${MINIO_ROOT_PASSWORD:-minioadmin}
export MINIO_CONSOLE_PORT=${MINIO_CONSOLE_PORT:-9001}
export MINIO_SERVER_PORT=${MINIO_SERVER_PORT:-9000}
export WEAVIATE_PORT=${WEAVIATE_PORT:-8080}
export WEAVIATE_GRPC_PORT=${WEAVIATE_GRPC_PORT:-50051}
export OLLAMA_PORT=${OLLAMA_PORT:-11434}

File: build/kill_server.sh [hunk 7]
```diff
@@ -13,7 +13,57 @@ echo "Project root: $PROJECT_ROOT"
...
+echo "MINIO_SERVER_PORT=$MINIO_SERVER_PORT"
+echo "WEAVIATE_PORT=$WEAVIATE_PORT"
+echo "WEAVIATE_GRPC_PORT=$WEAVIATE_GRPC_PORT"
+echo "OLLAMA_PORT=$OLLAMA_PORT"
+
 # Move to src/backend directory
 cd "$PROJECT_ROOT/src/backend" || exit 1
 echo "Bringing Docker down"
 docker-compose down
+
```
Comment (SaravanakumarR2018) [inline build/kill_server.sh:51]: Why are we pulling the docker files here?
Lets remove the following lines

# Pull required images
echo "Pulling required Docker images..."
docker-compose pull mysql_db
docker-compose pull minio
docker-compose pull ollama

# Pull weaviate image with retry
echo "Pulling weaviate image..."
for i in {1..5}; do
  docker-compose pull weaviate
  if [ $? -eq 0 ]; then
    break
  fi
  echo "Pull failed, retrying in 10 seconds..."
  sleep 10
done

echo "All required Docker images pulled successfully."

File: build/run_server.sh [hunk 8]
```diff
@@ -11,6 +11,37 @@ PROJECT_ROOT="$SCRIPT_DIR/.."  # Use the local directory structure
 export PROJECT_ROOT
 echo "PROJECT_ROOT is set to: $PROJECT_ROOT"
 
+# Set fallback values for critical environment variables
```
Comment (SaravanakumarR2018) [inline build/run_server.sh:14]: Lets source these values from .env file instead of the below way

export CHAT_SERVICE_PORT=${CHAT_SERVICE_PORT:-8000}
export DB_USER=${DB_USER:-root}
export DB_PASSWORD=${DB_PASSWORD:-admin123}
export DB_HOST=${DB_HOST:-mysql_db}
export MYSQL_DATABASE=${MYSQL_DATABASE:-mydatabase}
export MYSQL_PORT=${MYSQL_PORT:-3306}
export MINIO_ROOT_USER=${MINIO_ROOT_USER:-minioadmin}
export MINIO_ROOT_PASSWORD=${MINIO_ROOT_PASSWORD:-minioadmin}
export MINIO_CONSOLE_PORT=${MINIO_CONSOLE_PORT:-9001}
export MINIO_SERVER_PORT=${MINIO_SERVER_PORT:-9000}
export WEAVIATE_PORT=${WEAVIATE_PORT:-8080}
export WEAVIATE_GRPC_PORT=${WEAVIATE_GRPC_PORT:-50051}
export OLLAMA_PORT=${OLLAMA_PORT:-11434}

File: src/backend/chat_service/chat_service.py [hunk 9]
```diff
@@ -270,5 +270,7 @@ async def check_server_status(request: Request):
 # Run the server
 if __name__ == "__main__":
     import uvicorn
+    import os
 
-    uvicorn.run(app, host="0.0.0.0", port=8000)
+    port = int(os.environ.get("CHAT_SERVICE_PORT", 8000))
```
Comment (SaravanakumarR2018) [inline src/backend/chat_service/chat_service.py:275]: Convert this line without the default value so that we are sure the values are picked up from environmental variables
port = int(os.environ.get("CHAT_SERVICE_PORT", 8000))

File: src/backend/db/database_manager.py [hunk 10]
```diff
@@ -41,7 +41,7 @@ def _initialize_session_factory(self):
         logger.info(f"Connecting to database as user: {db_config['user']} on host: {db_config['host']}")
 
         engine = create_engine(
-            f"mysql+mysqlconnector://{db_config['user']}:{db_config['password']}@{db_config['host']}",
+            f"mysql+mysqlconnector://{db_config['user']}:{db_config['password']}@{db_config['host']}:{os.getenv('DB_PORT', '3306')}",
```
Comment (SaravanakumarR2018) [inline src/backend/db/database_manager.py:44]: Use it without the default value

File: src/backend/docker-compose.yml
Comment (SaravanakumarR2018) [inline src/backend/docker-compose.yml:1]: This file contains so many ports which are hard coded. Make sure all these ports are assigned through environmental values. Make sure none of the values have defaults

File: src/backend/docker-compose.yml [hunk 11]
```diff
@@ -118,5 +118,4 @@ volumes:
 
 # Define the absolute path for PROJECT_ROOT
 x-environment:
-  PROJECT_ROOT: ${PROJECT_ROOT}
```
Comment (SaravanakumarR2018) [inline src/backend/docker-compose.yml:121]: Remove this unnecessary change, in future make sure, all these unnecessary changes are not present

File: src/backend/minio/minio_manager.py [hunk 12]
```diff
@@ -14,10 +14,10 @@ class MinioManager:
     def __init__(self):
 
         try:
-            self.client=Minio(
-                os.getenv('MINIO_HOST','minio:9000'),
-                access_key=os.getenv('MINIO_ROOT_USER','minioadmin'),
-                secret_key=os.getenv('MINIO_ROOT_PASSWORD','minioadmin'),
+            self.client = Minio(
+                f"{os.getenv('MINIO_HOST', 'minio')}:9000",
```
Comment (SaravanakumarR2018) [inline src/backend/minio/minio_manager.py:18]: Make sure, you can change the below lines 

                f"{os.getenv('MINIO_HOST', 'minio')}:9000",

                
                like below
                
                minio: os.getenv('MINIO_SERVER_PORT')
                
                Do not use default values for the below lines
                
                                access_key=os.getenv('MINIO_ROOT_USER', 'minioadmin'),
                secret_key=os.getenv('MINIO_ROOT_PASSWORD', 'minioadmin'),

File: src/backend/minio/minio_manager.py [hunk 13]
```diff
@@ -117,4 +117,3 @@ def download_file(self,bucket_name,filename):
             else:
                 logger.error(f"Unexpected error during file download:{e}")
                 return {"error":f"An error occurred:{e}"}
-
```
Comment (SaravanakumarR2018) [inline src/backend/minio/minio_manager.py:120]: Remove this unnecessary change

File: src/backend/weaviate/weaviate_manager.py [hunk 14]
```diff
@@ -11,8 +11,9 @@ class WeaviateManager:
     def __init__(self):
 
         try:
+            weaviate_port = os.getenv('WEAVIATE_PORT', '8080')  # Default to 8080 if not set
             self.client = Client(
-                os.getenv('WEAVIATE_HOST', 'http://weaviate:8080')
```
Comment (SaravanakumarR2018) [inline src/backend/weaviate/weaviate_manager.py:15]: You dont need a weaviate_host and only get the weaviate port without the default value

File: test/IntegrationTests/test_addcustomer.py [hunk 15]
```diff
@@ -1,13 +1,15 @@
...
+import os
 
 # Set up logging configuration
 logging.basicConfig(level=logging.INFO)
 logger = logging.getLogger(__name__)
 
 class TestAddCustomerAPI(unittest.TestCase):
-    BASE_URL = "http://localhost:8000"
+    # Get the port from environment variables (default to 8000 if not set)
+    BASE_URL = f"http://localhost:{os.getenv('CHAT_SERVICE_PORT', 8000)}"
```
Comment (SaravanakumarR2018) [inline test/IntegrationTests/test_addcustomer.py:12]: Don't set default value for this environmental variable

**Review by SaravanakumarR2018 — COMMENTED** [review]
Create a .env file with the values and then load this file in shell script and github workflows so that we see consistent values in both shell script and github workflows

# .env file with default values

CHAT_SERVICE_PORT=8000
DB_USER=root
DB_PASSWORD=admin123
DB_HOST=mysql_db
MYSQL_DATABASE=mydatabase
MYSQL_PORT=3306
MINIO_ROOT_USER=minioadmin
MINIO_ROOT_PASSWORD=minioadmin
MINIO_CONSOLE_PORT=9001
MINIO_SERVER_PORT=9000
WEAVIATE_PORT=8080
WEAVIATE_GRPC_PORT=50051
OLLAMA_PORT=11434


Make sure in none of the places, we dont set default values for the environmental variables as these default variables will be obtained from the .env file

File: .github/workflows/integrationtests.yml (hunk 2 above)
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml:30]: Added a step to load the .env file

File: .github/workflows/integrationtests.yml (hunk 3 above)
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml:61]: I have removed the unnecessary change

File: .github/workflows/integrationtests.yml (hunk 4 above)
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml:107]: I have removed the unnecessary change

File: .github/workflows/integrationtests.yml (hunk 5 above)
Comment (santhiya-mj) [reply .github/workflows/integrationtests.yml]: The unnecessary change has been dropped

File: build/kill_server.sh (hunk 6 above)
Comment (santhiya-mj) [reply build/kill_server.sh]: Added .env file loading to the kill_server.sh script.

File: build/kill_server.sh (hunk 7 above)
Comment (santhiya-mj) [reply build/kill_server.sh]: I deleted the following lines from this file.

File: build/run_server.sh (hunk 8 above)
Comment (santhiya-mj) [reply build/run_server.sh]: Added .env file loading to the run_server.sh script.

File: src/backend/chat_service/chat_service.py (hunk 9 above)
Comment (santhiya-mj) [reply src/backend/chat_service/chat_service.py]: I replaced the default values with environmental variables.

File: src/backend/db/database_manager.py (hunk 10 above)
Comment (santhiya-mj) [reply src/backend/db/database_manager.py]: I removed the default values from this line

File: src/backend/docker-compose.yml
Comment (santhiya-mj) [reply src/backend/docker-compose.yml:1]: I removed hardcoded defaults and switched to environment variables for port configurations.

File: src/backend/minio/minio_manager.py (hunk 13 above)
Comment (santhiya-mj) [reply src/backend/minio/minio_manager.py:120]: The unnecessary change has been dropped

File: src/backend/weaviate/weaviate_manager.py (hunk 14 above)
Comment (santhiya-mj) [reply src/backend/weaviate/weaviate_manager.py:15]: I've modified all the changes as suggested. Removed the unnecessary Weaviate host and updated the Weaviate port to not include a default value.

File: test/IntegrationTests/test_addcustomer.py (hunk 15 above)
Comment (santhiya-mj) [reply test/IntegrationTests/test_addcustomer.py]: I removed default values from this code.
